include/st_string.h

```diff
@@ -311,6 +328,22 @@ namespace ST
         }
 #endif
 
+#ifdef ST_ENABLE_COREFOUNDATION
+        string(CFStringRef str,
+               utf_validation_t validation = ST_DEFAULT_VALIDATION)
+        {
+            set(str, validation);
+        }
+
+#ifdef __OBJC__
+        string(NSString* str,
+               utf_validation_t validation = ST_DEFAULT_VALIDATION)
+        {
+            set(ST_BRIDGED_CAST(CFStringRef)str, validation);
+        }
+#endif
```

Comment:
Thinking about it again, this on its own is already an ODR violation. This constructor will exist only when the header is included from an Objective-C++ file, but not when included from a regular C++ file. Linking those two files together would then lead to undefined behavior, as I understand it.

This makes at least the `NSString` part of the integration impossible. The `CFStringRef` integration would still be okay, but it's also much less convenient to use from Objective-C, which kind of kills the point of this integration... :(

Reply:
ugh, that's probably right...

and while `CFString` and `NSString` are toll-free bridged, they are not automatically cast to/from each other 😞

Reply:
It looks like it should be possible to avoid the ODR violation here by forward declaring NSString in two ways:

```objective-c++
#ifdef __OBJC__
    @class NSString;
#else
    class NSString;
#endif
```

Since we're just casting and not calling any Objective-C methods on NSString, that should allow un-conditionalizing the NSString variants.

This probably sounds like a ridiculous idea that couldn't possibly work, but it's what WebKit does in their public API: https://github.com/WebKit/WebKit/blob/b60e4db278efb963ae5a48fb81e0f71030263b12/Source/WTF/wtf/Compiler.h#L315-L323 🤷🏼‍♂️

Reply:
Yeah, "WTF" is about right :P I guess this is unofficially-officially supported, considering that Apple is heavily involved with both WebKit and clang.